sky/provision/azure/instance.py

```diff
@@ -39,6 +42,8 @@
 
 _RESOURCE_GROUP_NOT_FOUND_ERROR_MESSAGE = 'ResourceGroupNotFound'
 _POLL_INTERVAL = 1
+_LEGACY_NSG_NAME = 'ray-{cluster_name_on_cloud}-nsg'
```

Comment:
Could you add # TODO: We can remove this after 0.8.0 to ignore legacy nsg names.